R/get_ci.R

```diff
@@ -83,95 +167,297 @@ get_ci <- function(dist, interval, estimate, variance = NULL, denominator = NULL
+    lower = estimate * exp(-(stats::qnorm(p = 1 - (alpha / 2), mean = 0, sd = 1) * sqrt(variance_log))),
+    upper = estimate * exp(stats::qnorm(p = 1 - (alpha / 2), mean = 0, sd = 1) * sqrt(variance_log))
+  )
 
-    if (!all(variance >= 0, na.rm = TRUE)) {
-      stop("`variance` must be greater than or equal to 0")
-    }
+  return(result)
+}
 
-    if (length(estimate) != length(variance)) {
-      stop("length of `variance` must be equal to length of `estimate`")
-    }
 
-    if (!is.null(denominator)) {
-      stop("`denominator` must be NULL when `dist` is not 'poisson'")
-    }
+#' Calculate a Poisson distribution confidence interval
+#'
+#' @description Calculate a Poisson distribution confidence interval. This
+#' function imitates [stats::qpois()] using [stats::qchisq()] to allow for the
+#' continuous extension of the estimate.
+#'
+#' @param interval A scalar between 0 and 1, indicating the width of the
+#' interval. For example, use `0.95` to calculate a 95% confidence interval.
+#' @param x A numeric vector of positive integers. It can contain the
+#' stratum-specific number of events or the observed number of events in the
+#' study population.
+#' @param y A vector of positive values. It can contain the population-time for
+#' each stratum or the expected number of events. The default value is `1`.
+#'
+#' @details When working with rates, pass the stratum-specific
+#' number of events into `x` and the population-time for each stratum into `y`.
+#' When working with standardized morbidity/mortality ratios, pass the observed
+#' number of events into `x` and the expected number of events into `y`. Ensure
+#' that the values of `x` and `y` are aligned and that the vectors are the same
+#' length.
+#'
+#' To calculate a confidence interval around count data, pass a vector
+#' of counts into `x` and set `y = 1`. The shorter vector `y` will
+#' be repeated so that its length matches that of `x`.
+#'
+#' This function is used to add confidence intervals in [get_spec_rt()].
+#'
+#' @return A data frame with columns `lower` and `upper` which contain the lower
+#' and upper limits of a confidence interval respectively.
+#'
+#' @examples
+#' \dontrun{
+#' # calculate a single confidence interval
+#' get_ci_pois(interval = 0.95, x = 49, y = 52) # rate
+#' get_ci_pois(interval = 0.95, x = 11) # count
+#'
+#' # use columns of a data frame as inputs
+#' df <- data.frame(x = c(24, 49), y = c(94, 52))
+#'
+#' # using dplyr
+#' df %>%
+#'   dplyr::mutate(get_ci_pois(0.95, x, y))
+#'
+#' # using base
+#' get_ci_pois(0.95, df$x, df$y)
+#' }
+#'
+#' @references \href{http://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_stdrate_details.htm}{The STDRATE Procedure}
+#'
+#' @importFrom magrittr %>%
+#' @importFrom rlang .data
+#' @export
+get_ci_pois <- function(interval, x, y = 1) {
+  # TODO integrate Poisson CI into function that calculates SMR
+
+  # check validity of inputs
+
+  if (!is.numeric(interval)) {
```

Comment:
use the same error message with consistent wording that names all the requirements